agent/agent.go

```diff
@@ -1407,5 +1407,13 @@ func expandDirectory(dir string) (string, error) {
 		}
 		dir = filepath.Join(home, dir[1:])
 	}
-	return os.ExpandEnv(dir), nil
+
+	if !filepath.IsAbs(dir) && dir[0] != '$' {
```

Comment:
Let's say `dir = $USER/src` which would be a relative path (e.g. `mafredri/src`), it would not be turned into an absolute path in this case.

I still think it'd be simplest to expand dir before the abs check.

Reply:
let me try this. Thank you.